src/util/hooks.js

```diff
@@ -0,0 +1,23 @@
+/* @flow */
+
+export type YarnHook = 'resolveStep' | 'fetchStep' | 'linkStep' | 'buildStep';
```

Comment:
I would remove the `Step`in the next iteration since it doesn't add any value.

Reply:
It was like this at the beginning, then I considered that we might want to have a higher granularity of hooks later on. For example a hook that would be called once for each package we fetch.